AnimationGroup.lua

```diff
@@ -109,51 +130,46 @@ function AG:StopGroup(group)
             AG:Stop(animation)
         end
     end
-
     self:LoadProperties(group)
-
     group.playing = false
 end
 
 function AG:SaveProperties(group)
-    group.properties.alpha = group.parent:GetAlpha()
-    group.properties.width = group.parent:GetWidth()
-    group.properties.height = group.parent:GetHeight()
-    group.properties.point = { group.parent:GetPoint() }
+    for k, v in pairs(group:GetAnimations()) do
+        v:SaveProperties()
+    end
 end
 
 function AG:LoadProperties(group)
-    group.parent:SetAlpha(group.properties.alpha)
-    group.parent:SetWidth(group.properties.width)
-    group.parent:SetHeight(group.properties.height)
-
-    local point = group.properties.point
-    if point and point[1] then
-        group.parent:SetPoint(unpack(point))
+    for k, v in pairs(group:GetAnimations()) do
+        v:LoadProperties()
     end
 end
 
 function AG:Stop(animation)
     animation.time = 0
 
     if animation.OnUpdate then
-        animation:SetScript('OnUpdate', nil)
+        animation:_SetScript('OnUpdate', nil)
     end
 
     animation.playing = false
 end
 
 function AG:Play(animation)
-    if not animation.playing and animation.group.parent:IsVisible() then
+    if not animation.playing and animation.target:IsVisible() then
         animation.time = animation.group.reverse and animation.duration or 0
+        animation.startdelayTime = 0
+        animation.enddelayTime = 0
+        animation.progress = 0
+        animation.smoothProgress = 0
         animation.playing = true
-        animation:SetScript('OnUpdate', function() OnUpdate(this, arg1) end)
+        animation:_SetScript('OnUpdate', function() OnUpdate(this, arg1) end)
```

Comment:
Calling `_SetScript` directly - skips the assignment here: https://github.com/martinjlowm/AnimationGroup/blob/0b498e175bc3b65ec25c9fb74db2c756f8d0bddf/group.lua#L40

I don't remember what the purpose is with that though.

Reply:
animation:_SetScript('OnUpdate', function() OnUpdate(this, arg1) end)

here we must call SetScript of widget and not hooked one or will be mess if later in usercode some one will set another OnUpdate functiom.